litellm/llms/palm.py

```diff
@@ -118,17 +118,13 @@ def completion(
             k not in inference_params
         ):  # completion(top_k=3) > palm_config(top_k=3) <- allows for dynamic variables to be passed in
             inference_params[k] = v
-
-    prompt = ""
+    prompt = []
+    palm_messages = []     # Construct messages with roles
+    context = inference_params.get("context", None)
     for message in messages:
-        if "role" in message:
-            if message["role"] == "user":
-                prompt += f"{message['content']}"
-            else:
-                prompt += f"{message['content']}"
-        else:
-            prompt += f"{message['content']}"
-
+        role = message.get("role", "user")
+        palm_messages.append({"author": role, "content": message["content"]})
+        prompt.append({"context": context, "messages": palm_messages})
```

Comment:
@SujanShilakar  did you meant to add this line outside for loop?

Reply:
I just ran a quick check with 
```python
## dummy data
inference_params: dict = {'context': 'ABC'}
messages: list[dict] = [
    {'role': 'user', 'content': 'ABC'},
    {'role': 'system', 'content': 'EFG'},
    {'role': 'user', 'content': 'HIJ'},
    {'content': 'KLM'},
]

prompt = []
palm_messages = []     # Construct messages with roles
context = inference_params.get("context", None)
for message in messages:
    role = message.get("role", "user")
    palm_messages.append({"author": role, "content": message["content"]})
    prompt.append({"context": context, "messages": palm_messages})
print(prompt)
```

and got the following response
```
[{'context': 'ABC',
  'messages': [{'author': 'user', 'content': 'ABC'},
               {'author': 'system', 'content': 'EFG'},
               {'author': 'user', 'content': 'HIJ'},
               {'author': 'user', 'content': 'KLM'}]},
 {'context': 'ABC',
  'messages': [{'author': 'user', 'content': 'ABC'},
               {'author': 'system', 'content': 'EFG'},
               {'author': 'user', 'content': 'HIJ'},
               {'author': 'user', 'content': 'KLM'}]},
 {'context': 'ABC',
  'messages': [{'author': 'user', 'content': 'ABC'},
               {'author': 'system', 'content': 'EFG'},
               {'author': 'user', 'content': 'HIJ'},
               {'author': 'user', 'content': 'KLM'}]},
 {'context': 'ABC',
  'messages': [{'author': 'user', 'content': 'ABC'},
               {'author': 'system', 'content': 'EFG'},
               {'author': 'user', 'content': 'HIJ'},
               {'author': 'user', 'content': 'KLM'}]}
```

keeping the line out of for loop would result in `prompt=`
```
[{'context': 'ABC',
  'messages': [{'author': 'user', 'content': 'ABC'},
               {'author': 'system', 'content': 'EFG'},
               {'author': 'user', 'content': 'HIJ'},
               {'author': 'user', 'content': 'KLM'}]}]
```

or did you meant for `prompt`=
```
[{'context': 'ABC', 'messages': [{'author': 'user', 'content': 'ABC'}]},
 {'context': 'ABC', 'messages': [{'author': 'system', 'content': 'EFG'}]},
 {'context': 'ABC', 'messages': [{'author': 'user', 'content': 'HIJ'}]},
 {'context': 'ABC', 'messages': [{'author': 'user', 'content': 'KLM'}]}]```